### PR TITLE
op_mode: T6767: Check latest image version in VRF context for "add system image latest vrf <name>"

### PR DESCRIPTION
**Pull request by natali-rs1985**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [x] Other (please describe): change lisence file

## Related Task(s)

* https://vyos.dev/T6767
## Component(s) name

op mode
## How to test

```
set system update-check url https://raw.githubusercontent.com/vyos/vyos-rolling-nightly-builds/main/version.json
commit

vyos@vyos# run add system image  latest vrf TEST
Validating signature
Signature is valid
Validating image compatibility
Validating image checksums
What would you like to name this image? (Default: 1.5-rolling-202412060007) Would you like to set the new image as the default one for boot? [Y/n]
An active configuration was found. Would you like to copy it to the new image? [Y/n]
Copying configuration directory
Would you like to copy SSH host keys? [Y/n]
Copying SSH host keys
Copying system image files
Cleaning up
Unmounting target filesystems
Removing temporary files
...
```
## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: src/op_mode/image_installer.py [hunk 1]
```diff
@@ -550,11 +551,18 @@ def image_fetch(image_path: str, vrf: str = None,
...
     # Latest version gets url from configured "system update-check url"
     if image_path == 'latest':
-        config = ConfigTreeQuery()
-        if config.exists('system update-check url'):
-            configured_url_version = config.value('system update-check url')
-            remote_url_list = get_remote_version(configured_url_version)
-            image_path = remote_url_list[0].get('url')
+        if vrf is None:
+            config = ConfigTreeQuery()
+            if config.exists('system update-check url'):
```
Comment (c-po) [inline src/op_mode/image_installer.py]: Why duplicate the content of `/usr/libexec/vyos/latest-image-url.py` here and not reference it, too?

**Review by c-po — CHANGES_REQUESTED** [review]

**Comment by jestabro** [comment]
As mentioned on Slack, this should be rebased over current and pushed to sync with the changes in
https://github.com/vyos/vyos-1x/pull/4226
https://github.com/vyos/vyos-build/pull/824

A local build of this PR confirms a successful build.

**Comment by c-po** [comment]
@MergifyIo rebase

**Comment by jestabro** [comment]
I have not found a 'mergifyio rebase' to suffice in the past; this will likely need an explicit rebase and force push to the branch.

**Review by c-po — APPROVED** [review]

**Comment by c-po** [comment]
I wonder if it's a good idea to show REMOTE_USERNAME and REMOTE_PASSWORD on the CLI on error

```
The image cannot be fetched from: https://github.com/vyos/vyos-nightly-build/releases/download/1.5-rolling-202412100007/vyos-1.5-rolling-202412100007-generic-amd64.iso [Errno 1] failed to run command: REMOTE_USERNAME= REMOTE_PASSWORD=                 ip vrf exec red /usr/libexec/vyos/simple-download.py                 --local-file /tmp/vyos_installation.iso --remote-path https://github.com/vyos/vyos-nightly-build/releases/download/1.5-rolling-202412100007/vyos-1.5-rolling-202412100007-generic-amd64.iso
returned: HTTPSConnectionPool(host='github.com', port=443): Max retries exceeded with url: /vyos/vyos-nightly-build/releases/download/1.5-rolling-202412100007/vyos-1.5-rolling-202412100007-generic-amd64.iso (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fe011080a90>: Failed to establish a new connection: [Errno 16] Device or resource busy'))
```

**Review by c-po — CHANGES_REQUESTED** [review]
I wonder if it's a good idea to show REMOTE_USERNAME and REMOTE_PASSWORD on the CLI on error

```
The image cannot be fetched from: https://github.com/vyos/vyos-nightly-build/releases/download/1.5-rolling-202412100007/vyos-1.5-rolling-202412100007-generic-amd64.iso [Errno 1] failed to run command: REMOTE_USERNAME= REMOTE_PASSWORD=                 ip vrf exec red /usr/libexec/vyos/simple-download.py                 --local-file /tmp/vyos_installation.iso --remote-path https://github.com/vyos/vyos-nightly-build/releases/download/1.5-rolling-202412100007/vyos-1.5-rolling-202412100007-generic-amd64.iso
returned: HTTPSConnectionPool(host='github.com', port=443): Max retries exceeded with url: /vyos/vyos-nightly-build/releases/download/1.5-rolling-202412100007/vyos-1.5-rolling-202412100007-generic-amd64.iso (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fe011080a90>: Failed to establish a new connection: [Errno 16] Device or resource busy'))
```

**Review by dmbaturin — COMMENTED** [review]
I agree that this solution fixes the immediate problem. I still think we can do it better, though.

First, do we really need to execute the update check script as a subprocess? I think we could import it as a Python module and call a function from it instead.

Second, executing something in a VRF context is a need common enough that I believe we should add a named argument (`vrf`) to everything in `vyos.utils.subprocess`. That part should be a separate task, though.

**Review by dmbaturin — APPROVED** [review]
I don't mind merging it right now since it solves a problem but we need to address the deeper issue of VRF-awareness nonetheless.

**Review by c-po — APPROVED** [review]

**Pull request merged**